packages/shared/src/components/filters/BlockedWords.tsx

```diff
@@ -0,0 +1,131 @@
+      user,
+      RequestKey.UserBlocked,
+      {
+        entity: ContentPreferenceType.Word,
+      },
+    );
+    queryClient.invalidateQueries({
+      queryKey,
+    });
+  }, [queryClient, user]);
+  const queryResult = useBlockedQuery({
+    id: user.id,
+    entity: ContentPreferenceType.Word,
+  });
+  const flatBlockedWords =
+    queryResult?.data?.pages.flatMap(
+      (page) => page.edges.flatMap((edge) => edge.node) || [],
+    ) ?? [];
+  const onKeyDown = async (event: KeyboardEvent): Promise<void> => {
+    // Enter
+    if (event.keyCode === 13 && words?.length) {
+      await block({
+        id: words,
+        entity: ContentPreferenceType.Word,
+        entityName: words,
+      });
+      invalidateBlockedWords();
+      setWords('');
+    }
+  };
+
+  return (
+    <div className="flex flex-col gap-2">
+      <Typography tag={TypographyTag.H3} bold>
+        Blocked words
+      </Typography>
+      <Typography tag={TypographyTag.P} color={TypographyColor.Tertiary}>
+        Automatically filter out posts containing words you never want to see
+        again. Life’s too short for unnecessary noise.
+        {!isPlus ? (
+          <>
+            {' '}
+            To unlock this feature{' '}
+            <Link passHref href={plusUrl}>
+              <Typography
+                tag={TypographyTag.Link}
+                color={TypographyColor.Plus}
+                onClick={() => {
+                  logSubscriptionEvent({
+                    event_name: LogEvent.UpgradeSubscription,
+                    target_id: TargetId.BlockedWords,
+                  });
+                }}
+              >
+                upgrade to Plus
+              </Typography>
+            </Link>
+          </>
+        ) : undefined}
+      </Typography>
+      <TextField
+        inputId="blocked-words"
+        label="Add blocked words"
+        placeholder="Add blocked words"
+        hint="Add commas (,) to block multiple words. Press Enter to submit them."
+        hintIcon={<FeedbackIcon />}
+        onKeyDown={onKeyDown}
+        disabled={!isPlus}
+        value={words}
+        onChange={(event) => setWords(event.target.value)}
+        rightIcon={!isPlus ? <LockIcon /> : undefined}
+      />
+      <div className="flex flex-wrap gap-2">
+        {flatBlockedWords.map((word) => (
+          <GenericTagButton
+            key={word.referenceId}
+            className="group btn-tagBlocked"
+            icon={
+              <>
+                <BlockIcon className="ml-2 text-xl transition-transform group-hover:hidden" />
+                <MiniCloseIcon className="ml-2 hidden text-xl transition-transform group-hover:inline" />
+              </>
+            }
+            action={() =>
+              unblock({
+                id: word.referenceId,
+                entity: ContentPreferenceType.Word,
+                entityName: word.referenceId,
+              }).then(() => invalidateBlockedWords())
```

Comment:
Went for the invalidate route as we need to fully remove stuff and it seemed like a very niche system, but love to hear thoughts.
@capJavert

Reply:
Ideally we can modify local data, but if its hard this is not that bad.